swan-lake/learn-the-platform/test-document-the-code/test-ballerina-code/mocking.md

```diff
@@ -39,8 +39,37 @@ structurally equivalent to the real object via the mocking features in the test
 
 ***Example:***
 
-Let's make changes to the example in the [Test a simple function](/learn/test-ballerina-code/test-a-simple-function/) to define a 
-test double for the `clientEndpont` object.
+Consider the following example in which an `http:Client` interacts with an external endpoint to get a random joke.
+
+***main.bal***
+
+```ballerina
+import ballerina/io;
+import ballerina/http;
+import ballerina/regex;
+
+http:Client clientEndpoint = check new ("https://api.chucknorris.io/jokes/");
```

Comment:
Shall we make this final? This will get the function to be inferred as isolated.

Reply:
If we mark it as final, we can't assign mock objects directly and we'll have to use function mocking to set the mock object which might complicate the example IMO.

Reply:
Sorry, forgot. But, shouldn't that be our recommendation? Given that it won't work with isolated if not?

Reply:
we have specified how to mock `final` clients in https://github.com/ballerina-platform/ballerina-dev-website/pull/6322

This section aims to introduce the capabilities of object mocking.  When the variable is final, it requires a function mocking layer to set the mock object as it cannot be set directly. 
Maybe we can add a note to mention that we recommend using final clients and point to the `final mock` section. WDYT?

Reply:
If that's the case I guess it's OK to go with the current content.